helidon-linker/src/main/java/io/helidon/linker/util/FileUtils.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.linker.util;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * File utilities.
+ */
+public class FileUtils {
+    /**
+     * The Java Home directory for the running JVM.
+     */
+    public static final Path CURRENT_JAVA_HOME_DIR = Paths.get(System.getProperty("java.home"));
+
+    /**
+     * The working directory.
+     */
+    public static final Path WORKING_DIR = Paths.get(".").toAbsolutePath();
```

Comment:
You should use `System.getProperty("user.dir")` to get the "working" directory of a java program (AFAIK).

Reply:
Right, good catch.